Quick fix making more clear what --config means. (#1328)
o3de/o3de.org: content/docs/welcome-guide/setup/setup-from-github/building-linux.md

> 1. Use CMake to build the engine as an SDK, the same as if you installed the engine from an installer tool. The following example shows the `profile` build configuration.

Reviewer: this statement is no longer valid because the config now shows the 3 options.

Author: You do have a point.

    ```shell

>     cmake --build build/linux --target INSTALL --config profile -j <number of parallel build tasks>

Reviewer: For development profile is the recommended build type, debug is much slower to build and run and release is generally optimized for game releases.    If/when you build with the release config I'm not sure if tools like the editor will get built, what result did you get when you tested this?

Author: I didn't run --target INSTALL nor --config release so I don't know. I run with --target editor and --config debug.
I would just like to put a note that other build configurations exist and warning about "-Werror".

>     cmake --build build/linux --target INSTALL --config <debug|profile|release> -j <number of parallel build tasks>

Reviewer 2: I think we should keep the example for `profile`. I understand the purpose is to make it clear that O3DE supports different configurations, but the configuration recommended for this workflow is `profile`, and as a "Getting Started" topic, that's most likely enough info for newcomers to know. 

I very much agree that knowing the different configurations are useful, and that information is difficult to find. So in my next comments, I make some suggestions to address that. 
```suggestion
    cmake --build build/linux --target INSTALL --config profile -j <number of parallel build tasks>
```

    ```

    The `-j` is a recommended build tool optimization. It tells the Ninja build tool the number of parallel build tasks that will be executed simultaneously. We recommend that the 'number of parallel build tasks' matches the number of cores available on the Linux host machine.

>     The `--config` sets configuration type. This affects compilation flags. For example, `profile` turns on `-Werror`.

Reviewer 2: Mention the different build configuration types and link to the docs where the three are clearly defined. 
```suggestion
    The `--config` sets the build configuration type: `profile`, `debug`, or `release`. For setting up O3DE, `profile` is recommended. Read more on [O3DE's build configurations](/docs/user-guide/build/configure-and-build.md#generated-build-configurations).
```

If you accept this suggestion, can you also put it in the **Source engine** tab?

    Example:

> 

Reviewer 2: This comment is for line 119 (after "The engine takes a while to build..."), as I'm unable to make a comment there. 

In a "caution" callout box, we can mention the problem you encountered and a workaround. 
```suggestion

    {{< caution >}}
The `profile` configuration type enables the `-Werror` compilation flag. If you encounter build errors due to this, try building with the `debug` configuration instead. 
    {{< /caution >}}
    
```
----------------

Altogether, these suggested changes would render like this: 
![image](https://user-images.githubusercontent.com/75444793/152886013-4ba1af4b-02a9-4fdd-9fbe-3040b73b87b1.png)

Author: I have learned that every profile enables -Werror so this is not correct information.

See:
https://github.com/o3de/o3de/blob/development/cmake/Platform/Common/Clang/Configurations_clang.cmake

The problem I had was caused by git somehow getting commit where debug variable was unused in other profiles.
